### PR TITLE
BWS-96: + create permission api

### PR DESCRIPTION
**Pull request by bohdanshcherba**

### CODE REVIEW

**Review by what1s1ove — DISMISSED** [review]

**Review by Yaroslaww-1 — DISMISSED** [review]

**Review by ania-leichenko — DISMISSED** [review]

File: backend/src/api/eam/eam.api.ts [hunk 1]
```diff
@@ -98,6 +102,15 @@ const initEamApi: FastifyPluginAsync<Options> = async (fastify, opts) => {
       return rep.send(groups).status(HttpCode.OK);
     },
   });
+
+  fastify.route({
+    method: HttpMethod.GET,
+    url: `${EAMApiPath.GROUPS}${GroupsApiPath.PERMISSION}`,
```
Comment (what1s1ove) [inline backend/src/api/eam/eam.api.ts]:
```suggestion
    url: `${EAMApiPath.PERMISSION}`,
```

File: backend/src/services/group/group.service.ts [hunk 2]
```diff
@@ -38,8 +42,16 @@ class Group {
     if (groupByName) {
       throw new InvalidGroupNameError();
     }
+    if (!permissionIds) {
+      throw new PermissionSelectError();
+    }
```
Comment (what1s1ove) [inline backend/src/services/group/group.service.ts]:
```suggestion
```
we need to handle it by validation

File: shared/src/validation/eam-group/eam-group-create/group-create.validation-schema.ts [hunk 3]
```diff
@@ -22,6 +22,8 @@ const eamGroupCreate = Joi.object({
     .trim()
     .required(),
   [getNameOf<EAMGroupCreateRequestDto>('workersIds')]: Joi.array().required(),
+  [getNameOf<EAMGroupCreateRequestDto>('permissionIds')]:
+    Joi.array().required(),
```
Comment (what1s1ove) [inline shared/src/validation/eam-group/eam-group-create/group-create.validation-schema.ts]: should be at least 1 permission

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: backend/src/exceptions/permission-select-error/permission-select-error.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { HttpCode } from '~/common/enums/enums';
```
Comment (what1s1ove) [inline backend/src/exceptions/permission-select-error/permission-select-error.ts]: remove this file. handle it by validation

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: frontend/src/store/eam-group-configurate/actions.ts [hunk 5]
```diff
@@ -24,6 +24,7 @@ const create = createAsyncThunk<
     name: registerPayload.name,
     tenantId: tenant?.id ?? '',
     workersIds: registerPayload.workersIds,
+    permissionIds: [], // will be fix on frontend branch
```
Comment (what1s1ove) [inline frontend/src/store/eam-group-configurate/actions.ts:27]:
```suggestion
    permissionIds: [],
```
you also need to handle frontend part in this task

**Review by what1s1ove — DISMISSED** [review]

File: backend/src/services/permission/permission.service.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+class Permission {
+  #permissionRepository: typeof permissionRep;
+
+  constructor({ permissionRepository }: Constructor) {
+    this.#permissionRepository = permissionRepository;
+  }
+
+  async getAll(): Promise<EAMPermissionGetAllItemResponseDto[]> {
+    return this.#permissionRepository.getAll();
+  }
```
Comment (Yaroslaww-1) [inline backend/src/services/permission/permission.service.ts]:
```suggestion
  async getAll(): Promise<EAMPermissionGetAllResponse> {
    return { items: await this.#permissionRepository.getAll() };
  }
```

**Review by Yaroslaww-1 — CHANGES_REQUESTED** [review]

**Review by what1s1ove — APPROVED** [review]

**Review by Yaroslaww-1 — APPROVED** [review]

**Pull request merged**